controllers/articles.controller.js

```diff
@@ -27,9 +27,26 @@ exports.patchArticleById = (req, res, next) => {
 };
 
 exports.getArticles = (req, res, next) => {
-  selectArticles().then((articles) => {
-    res.status(200).send({ articles });
-  });
+  if (Object.keys(req.query).length > 0) {
+    if (
+      req.query.hasOwnProperty('sort_by') ||
+      req.query.hasOwnProperty('order_by') ||
+      req.query.hasOwnProperty('topic')
+    ) {
```

Comment:
This if statement on line 31 is slightly problematic. If someone did `?sort_by=author&grab_by=nothing&sortby=date` -> would we throw an error? From this if statement it feels like we'd be fine because it says if it has sort_by then it's fine but ideally (based on your testing) we'd want to still throw an error


Additionally, if your if statement is not actually doing anything, it's your else doing something, then I'd consider re-thinking the if statement. I'd rather you wrote one if statement that does the promise.reject than an unnecessary block that doesn't actual trigger an action to happen but is relied on to make a else block

Reply:
Fixed this logic and created another test for the more complex query you provided